### PR TITLE
[#9136] Change getBackUpIdentifier messages to make them more meaningful

### LINKED ISSUE #9136: EntityAttributes: Make getBackupIdentifier() message more meaningful
Ideally, `getBackupIdentifier()` should produce messages that will help developers to identify modified entities easily.

https://github.com/TEAMMATES/teammates/blob/98ef745d7a243f2330014a5ccbf059a3ee905a7a/src/main/java/teammates/common/datatransfer/attributes/AccountAttributes.java#L206-L209

The above is not helpful at all. 

In addition, I think we should name it as `getBackupMessage()` rather than `Identifier` as it is used as `log.info(entityToSaveAttributesForLogging.getBackupIdentifier());`

Another thing we can do is to push the method to `BaseEntity` instead as all DB manipulate entity instead entityAttribute. But, let's fix the message first before we move to the next step.

### PR DESCRIPTION
**Pull request by RonakLakhotia**
Fixes #9136 

**Outline of Solution**

I added a more meaningful message based on how the attribute is identified. Also wrote the test case.
I have only changed it for InstructorAtrributes at the moment. Wanted to get a review on my approach before I proceed.

### CODE REVIEW

**Comment by RonakLakhotia** [comment]
@xpdavid wanted your input on this

File: src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java [hunk 1]
```diff
@@ -142,12 +144,12 @@ public String getIdentificationString() {
...
     @Override
     public String getEntityTypeAsString() {
-        return "Instructor";
+        return DEFAULT_DISPLAY_NAME;
     }
 
     @Override
     public String getBackupIdentifier() {
-        return Const.SystemParams.COURSE_BACKUP_LOG_MSG + courseId;
+        return INSTRUCTOR_BACKUP_LOG_MSG + courseId + email;
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java]: There should be a delimiter between courseId and email.

File: src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java [hunk 2]
```diff
@@ -142,12 +144,12 @@ public String getIdentificationString() {
 
     @Override
     public String getEntityTypeAsString() {
-        return "Instructor";
+        return DEFAULT_DISPLAY_NAME;
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java]: I don't think `DEFAULT_DISPLAY_NAME` exists in every attribute.

Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java]: Should I create a new variable, "ATTRIBUTE_NAME" for each attribute in their respective classes?

File: src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java (hunk 1 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java]: Is it better to use the delimiter as a variable or simply append a "::" after courseId?

File: src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java (hunk 2 above)
Comment (xpdavid) [reply src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java]: Sounds good.

**Comment by RonakLakhotia** [comment]
Ready for review

**Comment by RonakLakhotia** [comment]
Ready for review

File: src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java [hunk 3]
```diff
@@ -245,12 +249,12 @@ public String getIdentificationString() {
 
     @Override
     public String getEntityTypeAsString() {
-        return "Feedback Question";
+        return ATTRIBUTE_NAME;
     }
 
     @Override
     public String getBackupIdentifier() {
-        return Const.SystemParams.COURSE_BACKUP_LOG_MSG + courseId;
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java]: Don't forget to delete `Const.SystemParams.COURSE_BACKUP_LOG_MSG` if is not used.

File: src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java [hunk 4]
```diff
@@ -142,7 +146,7 @@ public String getKey() {
      * Format: email%courseId e.g., adam@gmail.com%cs1101.
      */
     public String getId() {
-        return email + "%" + course;
+        return email + "::" + course;
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java]: You should not change this, it is not backward compatible.

**Comment by RonakLakhotia** [comment]
Ready for review

File: src/main/java/teammates/common/datatransfer/attributes/FeedbackResponseCommentAttributes.java [hunk 5]
```diff
@@ -22,6 +22,9 @@
  */
 public class FeedbackResponseCommentAttributes extends EntityAttributes<FeedbackResponseComment> {
 
+    private static final String FEEDBACK_RESPONSE_COMMENT_BACKUP_LOG_MSG = "Recently modified feedback response comment::";
+    private static final String ATTRIBUTE_NAME = "FeedbackResponseComment";
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/FeedbackResponseCommentAttributes.java]: Be consistent `Feedback Response Comment `

File: src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java [hunk 6]
```diff
@@ -176,6 +176,25 @@ public void testValidate() {
...
+    public void testGetBackUpIdentifier() {
+        FeedbackSessionAttributes sessionAttributes = FeedbackSessionAttributes
+                .builder("newFeedbackSessionName", "course", "email")
+                .withInstructions("default instructions")
+                .withCreatedTime(Instant.now())
+                .withStartTime(TimeHelperExtension.getInstantHoursOffsetFromNow(2))
+                .withEndTime(TimeHelperExtension.getInstantHoursOffsetFromNow(5))
+                .withSessionVisibleFromTime(TimeHelperExtension.getInstantHoursOffsetFromNow(1))
+                .withResultsVisibleFromTime(TimeHelperExtension.getInstantHoursOffsetFromNow(6))
+                .withTimeZone(ZoneId.of("Asia/Singapore")).withGracePeriodMinutes(0)
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java]: Be consistent. Break the line for each `with*`.

Comment (RonakLakhotia) [reply src/test/java/teammates/test/cases/datatransfer/FeedbackSessionAttributesTest.java]: My bad, will be more careful. 👍

File: src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java [hunk 7]
```diff
@@ -18,6 +18,9 @@
  */
 public class StudentProfileAttributes extends EntityAttributes<StudentProfile> {
 
+    private static final String STUDENT_PROFILE_BACKUP_LOG_MSG = "Recently modified student profile::";
+    private static final String ATTRIBUTE_NAME = "StudentProfile";
```
Comment (xpdavid) [inline src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java]: `Student Profile`?

File: src/test/java/teammates/test/cases/datatransfer/FeedbackResponseCommentAttributesTest.java [hunk 8]
```diff
@@ -113,4 +113,15 @@ public void testConvertCommentTextToStringForHtml() {
         String commentText = feedbackAttributes.getCommentAsHtmlString();
         assertEquals("hello Images Link: http:&#x2f;&#x2f;test.com&#x2f;test.png ", commentText);
     }
+
+    @Test
+    public void testGetBackUpIdentifier() {
+        FeedbackResponseCommentAttributes feedbackAttributes = FeedbackResponseCommentAttributes
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/datatransfer/FeedbackResponseCommentAttributesTest.java]: `commentAttributes` instead of `feedbackAttributes`?

Comment (RonakLakhotia) [reply src/test/java/teammates/test/cases/datatransfer/FeedbackResponseCommentAttributesTest.java]: Fixed

**Pull request merged**